wagtail/wagtailadmin/views/pages.py

```diff
@@ -588,16 +588,27 @@ def unpublish(request, page_id):
         raise PermissionDenied
 
     if request.method == 'POST':
+        include_descendants = request.POST.get("include_descendants", False)
+
         page.unpublish()
 
+        if include_descendants == 'True':
+            descendant_pages = page.get_descendants().live()
+            for descendant in descendant_pages:
+                descendant = descendant.specific
+                if descendant.permissions_for_user(request.user).can_unpublish():
```

Comment:
> As permissions in Wagtail can't be taken away as you go deeper in the tree, this check should never fail. Might be worth removing it

While this is true, I think we should keep all "knowledge" about permission logic contained within the permission_policy module, so it's a good thing for this code to perform the check rather than rely on that fact. It's entirely possible that we might make changes to the permission model in future, or make it configurable.